beater/server.go

```diff
@@ -220,10 +224,11 @@ func newGRPCServer(
 			model.ProcessBatchFunc(otlp.SetClientMetadata),
 			batchProcessor,
 		}
+		otlpConsumer.Processor = batchProcessor
```

Comment:
Sorry, I led you astray. I just took another look at the code, and we should be using a different consumer in `collectMetricsMonitoring` for HTTP and gRPC, since they are reported on different monitoring registries. I will push a change to your branch.